app/src/main/java/bg/zahov/app/ui/history/calendar/CalendarScreen.kt

```diff
@@ -0,0 +1,181 @@
+import com.kizitonwose.calendar.compose.HorizontalCalendar
+import com.kizitonwose.calendar.compose.rememberCalendarState
+import com.kizitonwose.calendar.core.CalendarDay
+import bg.zahov.fitness.app.R
+import com.kizitonwose.calendar.compose.CalendarState
+import com.kizitonwose.calendar.core.DayPosition
+import java.time.DayOfWeek
+import java.time.LocalDate
+import java.time.Month
+
+@Composable
+fun CalendarScreen(calendarViewModel: CalendarViewModel = viewModel()) {
+    val uiState by calendarViewModel.uiState.collectAsStateWithLifecycle()
+    val calendarState = rememberCalendarState(
+        startMonth = uiState.startMonth,
+        endMonth = uiState.endMonth,
+        firstVisibleMonth = uiState.endMonth,
+        firstDayOfWeek = uiState.firstDayOfWeek
+    )
+    CalendarContent(
+        calendarState = calendarState,
+        dayToHasUserWorkedOut = uiState.dayToHasUserWorkedOut,
+        daysOfWeek = uiState.daysOfWeek,
+        numberOfWorkoutsPerMonth = uiState.numberOfWorkoutsPerMonth
+    )
+}
+
+@Composable
+fun CalendarContent(
+    calendarState: CalendarState,
+    daysOfWeek: List<DayOfWeek>,
+    dayToHasUserWorkedOut: Map<LocalDate, Boolean>,
+    numberOfWorkoutsPerMonth: Map<Month, String>
+) {
+    HorizontalCalendar(
+        state = calendarState,
+        dayContent = { Day(it, dayToHasUserWorkedOut[it.date] == true) },
+        monthHeader = {
+            MonthHeader(
+                text = it.yearMonth.month.name,
+                daysOfWeek = daysOfWeek
+            )
+        },
+        monthFooter = {
+            MonthFooter(
+                workoutCount = stringResource(
+                    R.string.workouts_for_month,
+                    numberOfWorkoutsPerMonth[it.yearMonth.month] ?: 0
+                ),
+                year = it.yearMonth.year.toString(),
+            )
+        }
+    )
+}
+
+@Composable
+fun Day(
+    day: CalendarDay, hasUserWorkedOut: Boolean = true,
+    boxModifier: Modifier = Modifier
+        .aspectRatio(1f)
+        .background(
+            color = colorResource(R.color.background)
+        )
+        .padding(4.dp)
+) {
+    Box(
+        modifier = boxModifier,
+        contentAlignment = Alignment.Center
+    ) {
+        Text(
+            text = day.date.dayOfMonth.toString(),
+            color = if (day.position == DayPosition.MonthDate) Color.White else Color.Gray
+        )
+        if (hasUserWorkedOut) {
+            Icon(
+                painter = painterResource(R.drawable.ic_check_mark),
+                contentDescription = "",
+                modifier = Modifier
+                    .size(16.dp)
+                    .align(Alignment.TopEnd),
+                tint = Color.Unspecified
+            )
+        }
+    }
+}
+
+@Composable
+fun MonthHeader(
+    columnModifier: Modifier = Modifier.padding(top = 24.dp, bottom = 24.dp),
+    textModifier: Modifier = Modifier.padding(top = 8.dp, bottom = 4.dp),
+    daysModifier: Modifier = Modifier
```

Comment:
I would recommend to remove multiple modifiers. If you want some kind of reusability, provide content @Composable lambda. Also there is no need reusability to be so fragmentized.

Reply:
I thought it would be nice, however we don't need this to be reusable so I am going to remove it.